browser-test/src/admin/admin_program_visibility.test.ts

```diff
@@ -272,4 +273,54 @@ test.describe('Validate program visibility is correct for applicants and TIs', (
       'program-visibility-changes-all-ti-can-see-program',
     )
   })
+  test('create a program with disabled visibility, verify it is hidden from applicants and TIs', async () => {
+    const {page, tiDashboard, adminPrograms, applicantQuestions} = ctx
+    const programName = 'Disabled program'
+    await enableFeatureFlag(page, 'disabled_visibility_condition_enabled')
+
+    await test.step('login as a CiviForm admin and publish a disabled program', async () => {
+      await loginAsAdmin(page)
+
+      const programDescription = 'Description'
+      await adminPrograms.addProgram(
+        programName,
+        programDescription,
+        'https://usa.gov',
+        ProgramVisibility.DISABLED,
+      )
+      await adminPrograms.publishAllDrafts()
+    })
+
+    await test.step('log in as an applicant and verify the program is hidden from me', async () => {
+      // Login as applicant, verify program is hidden
```

Comment:
nitpick: you can delete this comment now since the context is represented in the `test.step` explanation